pkgs/by-name/ll/llama-cpp/package.nix

```diff
@@ -85,10 +86,9 @@ effectiveStdenv.mkDerivation (finalAttrs: {
       --replace '[bundle pathForResource:@"ggml-metal" ofType:@"metal"];' "@\"$out/bin/ggml-metal.metal\";"
   '';
 
-  nativeBuildInputs = [ cmake ninja pkg-config git ]
+  nativeBuildInputs = [ autoAddDriverRunpath cmake ninja pkg-config git ]
```

Comment:
Side note: I think we could keep it conditional. E.g. ROCm should work entirely without any impure drivers. OpenCL and Vulkan probably link the respective xxx-loader libraries which already know about `driverLink`.